chia/data_layer/data_store.py

```diff
@@ -484,6 +484,21 @@ async def get_first_generation(self, node_hash: bytes32, store_id: bytes32) -> O
 
             return int(row[0])
 
+    async def get_existing_hashes(self, node_hashes: list[bytes32], store_id: bytes32) -> set[bytes32]:
+        result: set[bytes32] = set()
+        batch_size = min(500, SQLITE_MAX_VARIABLE_NUMBER - 10)
+
+        for i in range(0, len(node_hashes), batch_size):
+            chunk = node_hashes[i : i + batch_size]
+            placeholders = ",".join(["?"] * len(chunk))
+            query = f"SELECT hash FROM nodes WHERE store_id = ? AND hash IN ({placeholders})"
```

Comment:
Awhile back I was encouraged to use `LIMIT` more.  I probably have slacked on that, but happen to remember right now...

```suggestion
            query = f"SELECT hash FROM nodes WHERE store_id = ? AND hash IN ({placeholders}) LIMIT {len(chunk)}"
```